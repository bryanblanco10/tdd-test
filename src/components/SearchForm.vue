<template>
  <div>
    <input type="text" v-model="inputData" />
    <button @click="fetchSpeaker(inputData)">Add speaker</button>
  </div>
</template>

<script>
import { mapActions } from "vuex";

export default {
  data() {
    return {
      inputData: "Speaker",
    };
  },
  methods: {
    ...mapActions({
      fetchSpeaker: "FETCH_SPEAKER",
    }),
  },
};
</script>

<style></style>
