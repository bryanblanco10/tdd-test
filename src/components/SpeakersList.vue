<template>
  <div>
    <speaker
      v-for="(speaker, index) in speakers"
      :key="index"
      :speaker="speaker"
    />
  </div>
</template>

<script>
import Speaker from "./Speaker";

export default {
  components: {
    Speaker,
  },
  props: {
    speakers: {
      type: Array,
      default: () => [],
    },
  },
};
</script>

<style></style>
