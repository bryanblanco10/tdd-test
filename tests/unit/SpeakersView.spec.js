import { shallowMount, createLocalVue } from "@vue/test-utils";
import SpeakersView from "@/views/SpeakersView";
import SearchForm from "@/components/SearchForm";
import SpeakersList from "@/components/SpeakersList";
import Vuex from "vuex";

/* Creating a localvue instance to be able to use vuex */
const localVue = createLocalVue();

/* Adding vuex to our new local Vue instance */
localVue.use(Vuex);

let speaker = {
  name: "Bryan",
};

describe("SpeakersView", () => {
  /* Creating global variables to access throung the test */
  let store, getters;

  /* executes beforeEach Test */
  beforeEach(() => {
    /* it creates a store with the desired getters */
    getters = {
      speakers: () => [speaker, speaker],
    };
    store = new Vuex.Store({
      getters,
    });
  });

  const build = () => {
    const wrapper = shallowMount(SpeakersView, {
      /* Passing the created store and the vue instance when mounting the component */
      store,
      localVue,
    });

    return {
      wrapper,
      searchForm: () => wrapper.findComponent(SearchForm),
      speakersList: () => wrapper.findComponent(SpeakersList),
    };
  };

  it("renders the component", () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render the correct", () => {
    const { searchForm, speakersList } = build();

    expect(searchForm().exists()).toBeTruthy();
    expect(speakersList().exists()).toBeTruthy();
  });

  it("should passes speakers to the speakersList", () => {
    const { speakersList } = build();
    expect(speakersList().vm.speakers).toEqual([speaker, speaker]);
    expect(speakersList().vm.speakers).toBe(store.getters.speakers);
  });
});
