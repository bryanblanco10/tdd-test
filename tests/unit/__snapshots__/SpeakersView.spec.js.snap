// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeakersView renders the component 1`] = `
<div>
  <search-form-stub></search-form-stub>
  <speakers-list-stub speakers="[object Object],[object Object]"></speakers-list-stub>
</div>
`;
